### DISCUSSION TITLE
Detect changes in underlying dependencies

### DISCUSSION THREAD

**Discussion by david1542** [python-poetry/poetry#7537] [Q&A]
Hi everyone,

Is there a way to detect differences between the `poetry.lock` dependencies and the dependencies that are actually installed?
The motivation for this is due to the fact I work with lots of virtual environments, and sometimes these environments change (due to newer commits). I was wondering if there is an easy way to detect changes using poetry, so that I'd know when to run `poetry install` again.

This feature exists in WebStorm + Yarn / NPM, for example.

**Comment by radoering** [comment]
I suppose `poetry install --dry-run` comes closest.
